### PR TITLE
arm64: Implement saturating SIMD arithmetic

### PR DESCRIPTION
**Pull request by jgouly**
Copyright (c) 2020, Arm Limited.

### CODE REVIEW

File: cranelift/codegen/src/isa/aarch64/inst/mod.rs [hunk 1]
```diff
@@ -209,12 +209,16 @@ pub enum VecExtendOp {
 pub enum VecALUOp {
     /// Signed saturating add
     SQAddScalar,
+    Sqadd,
```
Comment (cfallin) [inline cranelift/codegen/src/isa/aarch64/inst/mod.rs:212]: Tiniest possible nit, but: we should be consistent with capitalization, so either `SQAdd` here, or change above to `SqaddScalar`.

File: cranelift/codegen/src/isa/aarch64/lower_inst.rs [hunk 2]
```diff
@@ -97,70 +97,108 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
...
-            let alu_op = if is_signed {
-                VecALUOp::SQSubScalar
-            } else {
-                VecALUOp::UQSubScalar
-            };
-            let va = ctx.alloc_tmp(RegClass::V128, I128);
-            let vb = ctx.alloc_tmp(RegClass::V128, I128);
-            let ra = put_input_in_reg(ctx, inputs[0], narrow_mode);
-            let rb = put_input_in_reg(ctx, inputs[1], narrow_mode);
+            let ty = ty.unwrap();
```
Comment (cfallin) [inline cranelift/codegen/src/isa/aarch64/lower_inst.rs:153]: Pre-existing (my fault, sorry), but the subtract case is basically redundant with above, modulo the opcode itself -- could we merge them?

File: cranelift/codegen/src/isa/aarch64/inst/mod.rs (hunk 1 above)
Comment (jgouly) [reply cranelift/codegen/src/isa/aarch64/inst/mod.rs:212]: @akirilov-arm has a patch (not yet on github) that is going to be renaming/moving the Scalar version. Then `Sqadd` will be consistent with the other `VecALUOp` variants.

Comment (cfallin) [reply cranelift/codegen/src/isa/aarch64/inst/mod.rs:212]: Sounds good, thanks.

**Comment by cfallin** [comment]
Thanks!

**Pull request merged**